### ISSUE TITLE
multicov not always correct

### ISSUE THREAD

**Issue by brendanofallon** [arq5x/bedtools2#498]
Looks like the behavior of bedtools multicov is not always correct. Specifically the read counts for some regions are very seldomly calculated incorrectly. Here's the behavior we've noticed:
 1. Start with some bam file and use samtools to modify in some trivial way - for instance, filter some reads out and dump them to another file (using -U), then merge the reads back in to the filtered file to produce a file equivalent to the original.
 2. Count reads on the "before" and "after" bams using multicov. They should be identical, and tools such as GATK DepthOfCoverage and samtools / pysam confirm this (at least, I haven't seem them get it wrong yet). However, for multicov, at some tiny fraction of sites, typically around 0.01-0.1%, the read counts aren't identical. Yikes! 

 We noticed this when trying to simulate CNVs using samtools and bedtools to generate read depth data. It's often just a few targets out of 10,000's, but it's enough to generate false positive CNV calls for us. 
 Happy to supply before / after BAMs and regions that aren't working.

**Comment by arq5x** [comment]
Could you provide a minimal example so I can debug?

**Comment by brendanofallon** [comment]
Minimal examples are difficult since the events are very rare and hence typically require large alignments to generate a few definitive observations. I'd be happy to supply a larger .bam however, along with locations of what seem to be affected regions. I have access to the CHPC at the U of U, so I could easily place the file there if you wish. Alternatively, I could FTP it to a location of your choice. Dropbox / related tools are difficult at my work location.

**Comment by arq5x** [comment]
sure, CHPC location and a specific locus that appears to be wrong would be great!

**Comment by brendanofallon** [comment]
OK, please take a look at the files in the directory /uufs/chpc.utah.edu/common/home/u0379426/suspicious_multicov

..there's a README, example .bam, two BED files each with a single affected regions, and an example shell script demonstrating some of the behavior. 

Suspiciously, using samtools view to extract the affected regions into a smaller bam changes the bedtools result back to a value that seems correct (at least, one that closely corresponds to samtools view -c), hence the big bam.

**Comment by brendanofallon** [comment]
Howdy, just following up on this. Were you able to replicate the issue? Let me know if I can provide any more information.

**Comment by arq5x** [comment]
Sorry, I haven't had the bandwidth to tackle this yet.

**Comment by ghuls** [comment]
@brendanofallon 
Did you try to make a minimal BAM file, by using samtools view for a certain region only?

**Comment by brendanofallon** [comment]
Yes, please see above comments. Extracting a smaller region changes the behavior. Is there an issue  using the larger bam that I shared earlier?

**Comment by ghuls** [comment]
I can't access that link (as it is not public).
